### PR TITLE
CLN: assorted, mostly typing

### PR DESCRIPTION
**Pull request by jbrockmendel**
diff_2d no longer needs to be in the pxi.in file, so moved it to the pyx

A couple of recently-identified bugs in the groupby code are caused by passing incorrect types, so im getting more motivated to add annotations in/around the affected code.

### CODE REVIEW

File: pandas/core/groupby/generic.py [hunk 1]
```diff
@@ -971,7 +972,7 @@ def _cython_agg_blocks(self, how, alt=None, numeric_only=True, min_count=-1):
             locs = block.mgr_locs.as_array
             try:
                 result, _ = self.grouper.aggregate(
-                    block.values, how, axis=agg_axis, min_count=min_count
+                    block.values, how, axis=1, min_count=min_count
```
Comment (gfyoung) [inline pandas/core/groupby/generic.py]: How come we can hardcode this value?

**Review by gfyoung — COMMENTED** [review]
I think all of these changes look okay, but I also feel that this PR covers a little too much ground and could have benefited from being broken up into smaller parts (e.g. typing, Cython refactoring).

I somewhat on the fence, so would be good to get some other eyes to look at this.

**Comment by jbrockmendel** [comment]
> but I also feel that this PR covers a little too much ground and could have benefited from being broken up into smaller parts (e.g. typing, Cython refactoring

Totally reasonable.  If others find this a deal-breaker I'll split this up.  Otherwise I'll apply this advice in future CLN PRs

**Comment by jreback** [comment]
this is ok, but generally yes would like to split largish changes. note to @WillAyd this might conflict with your groupby PR?

**Pull request merged**